src/XMakeCommandLine/XMake.cs

```diff
@@ -226,6 +226,21 @@ private static void HandleConfigurationException(Exception ex)
         }
 
         /// <summary>
+        /// Insert the command executable path as the first element of the args array.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static string[] ConstructArrayArg(string[] args)
```

Comment:
@AndyGerlicher I gave Mihai the opposite feedback yesterday.  A command line as a string is not a cross-platform concept- on Linux-like OS's apparently the process APIs take an array of arguments, and it's up to the command interpreter to handle parsing the command line into arguments.  Also, if we were to construct the command line as a string here, it's just going to be broken back down into an argument array anyway, and we'd have to make sure to handle all the quoting and escaping so that it would correctly round-trip.